source/git/source.go

```diff
@@ -639,9 +646,11 @@ func (gs *gitSourceHandler) Snapshot(ctx context.Context, g session.Group) (out
 	}
 
 	git = git.New(gitutil.WithWorkTree(cd), gitutil.WithGitDir(gitDir))
-	_, err = git.Run(ctx, "submodule", "update", "--init", "--recursive", "--depth=1")
-	if err != nil {
-		return nil, errors.Wrapf(err, "failed to update submodules for %s", urlutil.RedactCredentials(gs.src.Remote))
+	if !gs.src.SkipSubmodules {
+		_, err = git.Run(ctx, "submodule", "update", "--init", "--recursive", "--depth=1")
+		if err != nil {
+			return nil, errors.Wrapf(err, "failed to update submodules for %s", urlutil.RedactCredentials(gs.src.Remote))
+		}
```

Comment:
Maybe that's an edge case but was wondering if we could have `submodules` being either a bool to enable/disable init submodules but also a list to init some of them only like `submodules=./sub1,./sub2`?

Reply:
Possible problems:
- more idiomatic querystring would be `submodules=./sub1&submodules=./sub2`
- what if submodule path is `true` or `false`
- should this support pattern matching


Because of this would make more sense as a different builtin key that can be added later in a follow-up?

Reply:
> Because of this would make more sense as a different builtin key that can be added later in a follow-up?

Mostly looking at reusing the same key but the query form you suggest `submodules=./sub1&submodules=./sub2` makes more sense. So yeah we might need another key.

What do you think of `init-submodules=<true|false>` default to `true` and another one `filter-submodules=<pattern>`?

Reply:
Yes, was thinking smth like `submodules-filter`. Don't know if we would need `init-` prefix

Reply:
Yes `submodules=<bool>` and `submodules-filter=<pattern>` LGTM